chitchat_server/models.py

```diff
@@ -36,18 +39,60 @@ class User(db.Model):
     admin: Mapped[bool]
     password: Mapped[str]
 
+    def __init__(self, email, name, admin, password_plaintext):
+        self.email = email
+        self.name = name
+        self.admin = admin
+        salt = secrets.token_hex(8)
+        salted_pass = salt[:8] + password_plaintext + salt[8:]
+        self.password = salt+hashlib.sha256(salted_pass.encode('ascii')).hexdigest()
+
+    def check_password(self, plaintext_password):
+        salt, hash = self.password[:16], self.password[16:]
+        salted_pass = salt[:8] + plaintext_password + salt[8:]
+        check_hash = hashlib.sha256(salted_pass.encode('ascii')).hexdigest()
+        return check_hash == hash
+    
+    def to_json(self):
+        column_map = {
+                "email": self.email,
+                "name": self.name,
+                "id": self.id
+        }
+        return json.dumps(column_map)
+
 class Post(db.Model):
     __tablename__ = "posts"
     id: Mapped[int] = mapped_column(primary_key=True)
     contents: Mapped[str]
     user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
     created_at: Mapped[datetime]
 
+    def to_json(self):
```

Comment:
Good json file this wraps it up in a nice packet for usage later on on the paths